packages/interactivity/src/init.js

```diff
@@ -21,15 +21,25 @@ export const getRegionRootFragment = ( region ) => {
 	return regionRootFragments.get( region );
 };
 
+function yieldToMain() {
+	return new Promise( ( resolve ) => {
+		// TODO: Use scheduler.yield() when available.
+		setTimeout( resolve, 0 );
+	} );
+}
+
 // Initialize the router with the initial DOM.
 export const init = async () => {
-	document
-		.querySelectorAll( `[data-${ directivePrefix }-interactive]` )
-		.forEach( ( node ) => {
-			if ( ! hydratedIslands.has( node ) ) {
-				const fragment = getRegionRootFragment( node );
-				const vdom = toVdom( node );
-				hydrate( vdom, fragment );
-			}
-		} );
+	const nodes = document.querySelectorAll(
+		`[data-${ directivePrefix }-interactive]`
+	);
+
+	for ( const node of nodes ) {
+		if ( ! hydratedIslands.has( node ) ) {
+			await yieldToMain();
+			const fragment = getRegionRootFragment( node );
+			const vdom = toVdom( node );
+			hydrate( vdom, fragment );
```

Comment:
If necessary, these two tasks could also be broken down:

```suggestion
			const vdom = toVdom( node );
			await yieldToMain();
			hydrate( vdom, fragment );
```

Reply:
It looks like `toVdom` takes about a third of the amount of time as `hydrate`. But breaking it up here will further ensure that there are no long tasks:

![image](https://github.com/WordPress/gutenberg/assets/134745/b9aa0131-dc6b-4e7f-96fe-c52b98e46c4f)